astropy/tests/pytest_plugins.py

```diff
@@ -1,281 +0,0 @@
-
-    s += "Running tests in {0}.\n\n".format(" ".join(dirs))
-
-    s += "Date: {0}\n\n".format(datetime.datetime.now().isoformat()[:19])
-
-    from platform import platform
-    plat = platform()
-    if isinstance(plat, bytes):
-        plat = plat.decode(stdoutencoding, 'replace')
-    s += "Platform: {0}\n\n".format(plat)
-    s += "Executable: {0}\n\n".format(sys.executable)
-    s += "Full Python Version: \n{0}\n\n".format(sys.version)
-
-    s += "encodings: sys: {0}, locale: {1}, filesystem: {2}".format(
-        sys.getdefaultencoding(),
-        locale.getpreferredencoding(),
-        sys.getfilesystemencoding())
-    s += '\n'
-
-    s += "byteorder: {0}\n".format(sys.byteorder)
-    s += "float info: dig: {0.dig}, mant_dig: {0.dig}\n\n".format(
-        sys.float_info)
-
-    for module_display, module_name in PYTEST_HEADER_MODULES.items():
-        try:
-            with ignore_warnings(DeprecationWarning):
-                module = resolve_name(module_name)
-        except ImportError:
-            s += "{0}: not available\n".format(module_display)
-        else:
-            try:
-                version = module.__version__
-            except AttributeError:
-                version = 'unknown (no __version__ attribute)'
-            s += "{0}: {1}\n".format(module_display, version)
-
-    special_opts = ["remote_data", "pep8"]
-    opts = []
-    for op in special_opts:
-        op_value = getattr(config.option, op, None)
-        if op_value:
-            if isinstance(op_value, str):
-                op = ': '.join((op, op_value))
-            opts.append(op)
-    if opts:
-        s += "Using Astropy options: {0}.\n".format(", ".join(opts))
-
-    return s
-
-
-def pytest_pycollect_makemodule(path, parent):
-    # This is where we set up testing both with and without
-
-    return pytest.Module(path, parent)
-
-
-class Pair(pytest.File):
-    """
-    This class treats a given test .py file as a pair of .py files
-    where one has __future__ unicode_literals and the other does not.
-    """
-
-    def collect(self):
-        # First, just do the regular import of the module to make
-        # sure it's sane and valid.  This block is copied directly
-        # from py.test
-        try:
-            mod = self.fspath.pyimport(ensuresyspath=True)
-        except SyntaxError:
-            import py
-            excinfo = py.code.ExceptionInfo()
-            raise self.CollectError(excinfo.getrepr(style="short"))
-        except self.fspath.ImportMismatchError:
-            e = sys.exc_info()[1]
-            raise self.CollectError(
-                "import file mismatch:\n"
-                "imported module {!r} has this __file__ attribute:\n"
-                "  {}\n"
-                "which is not the same as the test file we want to collect:\n"
-                "  {}\n"
-                "HINT: remove __pycache__ / .pyc files and/or use a "
-                "unique basename for your test file modules".format(e.args))
```

Comment:
Just a question about this one: My static analysis checker marks this as "Error" because the string has 3 placeholders but it only get's one argument. If this is relocated it would be good to fix it.

Reply:
I think it's probably intended to be `.format(*e.args)`. Does that make sense?

Reply:
I think so too, but if it's deleted nevertheless (I couldn't find it as added somewhere else in the PR) we don't need to bother.

Reply:
Yes I think it has been removed entirely since it was part of the unicode plugin which is no longer needed.

Reply:
ok, in that case ignore my comment :)